ambry-account/src/main/java/com/github/ambry/account/InMemoryUnknownAccountService.java

```diff
@@ -26,6 +27,7 @@
 class InMemoryUnknownAccountService implements AccountService {
   private static final Collection<Account> accounts =
       Collections.unmodifiableCollection(Collections.singletonList(Account.UNKNOWN_ACCOUNT));
+  private final CopyOnWriteArraySet<AccountUpdateListener> accountUpdateListeners = new CopyOnWriteArraySet<>();
```

Comment:
So `CopyOnWriteArraySet` is optimized for cases when you iterate more than adding/removing from the set?